Removed synchronized keyword from native methods

kfilio opened the pull request:
- cherry-picked commit a784e7817258e51eb677318db6da4d14cd531bea
- removed synchronized key word from native methods in NativeDB.java
- added method limit() back into DB.java

mruffing committed:
Removed synchronized keyword from native methods

The default sqlite thread mode is serialized.
The underlying C sqlite library is already providing thread safety (via serialized mode).
Removing these extra mutex/locks improves concurrency.

Updated pom to include the source plugin

mruffing commented on src/main/java/org/sqlite/core/DB.java:
```diff
@@ -695,6 +695,15 @@ public final synchronized int finalize(CoreStatement stmt) throws SQLException {
      */
     public abstract int restore(String dbName, String sourceFileName, ProgressObserver observer) throws SQLException;
 
+    /**
```
Is this new?

kfilio replied:
Yes this is new [DB.java](https://github.com/xerial/sqlite-jdbc/blob/3.32.3.2/src/main/java/org/sqlite/core/DB.java) line 696

kfilio replied:
Nope, I was looking at the snapshot for 3.32.3.3, the other method in NativeDB.java is also related to the newest snapshot

mruffing commented on src/main/java/org/sqlite/core/NativeDB.java:
```diff
@@ -16,15 +16,11 @@
 package org.sqlite.core;
 
-import org.sqlite.BusyHandler;
+import org.sqlite.*;
```
Update this to the original imports.

mruffing commented on src/main/java/org/sqlite/core/NativeDB.java:
```diff
@@ -160,276 +156,279 @@ public synchronized String libversion() {
+    native void free_functions();
+
+    @Override
+    public native int limit(int id, int value) throws SQLException;
```
Was this method removed in 3.32.3.2?

kfilio committed:
Removed extra methods from DB.java and NativeDB.java

- removed limit() from NativeDB.java
- removed restore() from DB.java
- fixed imports in NativeDB.java

kfilio committed:
Added new lines back into end of file

mruffing commented on src/main/java/org/sqlite/core/DB.java:
```diff
@@ -694,8 +694,8 @@ public final synchronized int finalize(CoreStatement stmt) throws SQLException {
     public abstract int restore(String dbName, String sourceFileName, ProgressObserver observer) throws SQLException;
-
-
+    
```
Remove this edit.

kfilio committed:
Removed changes from DB.java
